blockmanager.go

```diff
@@ -3044,3 +3091,167 @@ func (l *lightHeaderCtx) RelativeAncestorCtx(
+	if b.sideLoadBlkHdrReader.HeadersChain() != b.cfg.ChainParams.Net {
+		log.Error("headers from side load file are of network %v "+
+			"and so incompatible with neutrino's current bitcoin network "+
+			"-- skipping side loading", b.sideLoadBlkHdrReader.HeadersChain())
+
+		return
+	}
+	log.Infof("Side loading headers from %v to %v", b.headerTip, b.sideLoadBlkHdrReader.EndHeight())
+
+	// Set headerTip to enable reader supply header, node needs
+	err := b.sideLoadBlkHdrReader.SetHeight(int64(b.headerTip))
+	if err != nil {
+		log.Errorf("error while setting height for sideload--- skipping sideloading: "+
+			"%v", err)
+
+		return
+	}
+
+	// Start batch writer to write headers into DB at specified batch size and ticker duration.
+	b.blockHeaderBatchWriter.Start()
+	defer b.blockHeaderBatchWriter.Stop()
+	for {
+		select {
+		case <-b.quit:
+			return
+		default:
+			// Request header
+			header, headerErr := b.sideLoadBlkHdrReader.NextHeader()
+			// If any error occurs while fetching headers that does not indicate
+			// an end of file, return immediately.
+			if headerErr != nil {
+				if headerErr == chaindataloader.ErrEndOfFile {
+					log.Infof("Successfully completed sideloading")
+					return
+				}
+
+				log.Errorf("error while fetching headers -- skipping sideloading %v", err)
+				return
+			}
+
+			var (
+				node     *headerlist.Node
+				prevNode *headerlist.Node
+			)
+			// Ensure there is a previous header to compare against.
+			prevNodeEl := b.headerList.Back()
+			if prevNodeEl == nil {
+				log.Warnf("side load - Header list does not contain a previous" +
+					"element as expected -- exiting side load")
+
+				return
+			}
+
+			node = &headerlist.Node{Header: *header}
+			prevNode = prevNodeEl
+			node.Height = prevNode.Height + 1
+
+			if !b.cfg.sideLoad.SkipVerify {
+				valid, err := b.verifyBlockHeader(header, *prevNode)
+				if err != nil || !valid {
+					log.Debugf("Side Load- Did not pass verification at height %v"+
+						"-- rolling back to last verified checkpoint and skipping sideload", node.Height)
+
+					prevCheckpoint := b.findPreviousHeaderCheckpoint(
+						node.Height,
+					)
+
+					log.Infof("Rolling back to previous validated "+
+						"checkpoint at height %d/hash %s",
+						prevCheckpoint.Height,
+						prevCheckpoint.Hash)
+
+					// It is possible that the number of headers in the batch writer's buffer is less than the
+					// BlkHeaderWriteBatch and so these headers have not been written into the DB at this time.
+					// We wait for the DBTickerBlkHeaderWriteDuration to expire and write these headers into the
+					// DB before we roll back which should take approximately twice the DBTickerBlkHeaderWriteDuration.
+					// This ensures that these headers are wiped off, during the rollback, in the batch writer's buffer
+					// and in the DB as well.
+					time.Sleep(DBTickerBlkHeaderWriteDuration * 2)
+					err = b.rollBackToHeight(uint32(prevCheckpoint.Height))
+					if err != nil {
```

Comment:
No panics.

Reply:
I was just following the design here:
https://github.com/lightninglabs/neutrino/blob/76952b79ac2a0c77450bc4dbd74275c325f5b1ef/blockmanager.go#L2669C1-L2673C5

Reply:
But I would just remove that